app/models/manageiq/providers/oracle_cloud/container_manager.rb

```diff
@@ -0,0 +1,115 @@
+  require_nested :Container
+  require_nested :ContainerGroup
+  require_nested :ContainerNode
+  require_nested :EventCatcher
+  require_nested :EventParser
+  require_nested :Refresher
+  require_nested :RefreshWorker
+
+  def connect_options(options = {})
+    options.merge(
+      :hostname    => options[:hostname] || address,
+      :port        => options[:port] || port,
+      :bearer      => bearer_token(options[:auth_type]),
+      :http_proxy  => self.options ? self.options.fetch_path(:proxy_settings, :http_proxy) : nil,
+      :ssl_options => options[:ssl_options] || {
+        :verify_ssl => verify_ssl_mode,
+        :cert_store => ssl_cert_store
+      }
+    )
+  end
+
+  def bearer_token(auth_type)
+    authentication = authentication_best_fit(auth_type || "bearer")
+    self.class.bearer_token(realm, authentication.userid, authentication.auth_key, authentication.public_key, provider_region, uid_ems)
+  end
+
+  class << self
+    def ems_type
+      @ems_type ||= "oke".freeze
+    end
+
+    def description
+      @description ||= "Oracle Kubernetes Engine".freeze
+    end
+
+    def display_name(number = 1)
+      n_('Container Provider (Oracle)', 'Container Providers (Oracle)', number)
+    end
+
+    def default_port
+      443
+    end
+
+    def params_for_create
+      {
+      }
+    end
+
+    def verify_credentials(args)
+      region, tenant, cluster_id = args.values_at("provider_region", "realm", "uid_ems")
+
+      default_endpoint = args.dig("endpoints", "default")
+      hostname, port = default_endpoint&.values_at("hostname", "port")
+
+      default_authentication = args.dig("authentications", "default")
+      user, private_key, public_key = default_authentication&.values_at("userid", "auth_key", "public_key")
+      private_key ||= find(args["id"]).authentication_token("default")
+
+      bearer = bearer_token(tenant, user, private_key, public_key, region, cluster_id)
+
+      options = {
+        :bearer      => bearer,
+        :ssl_options => {
+          :verify_ssl => OpenSSL::SSL::VERIFY_NONE
+        }
+      }
+
+      !!raw_connect(hostname, port, options)
+    end
+
+    def bearer_token(tenant, user, private_key, public_key, region, cluster_id)
+      config                  = oci_config(tenant, user, private_key, public_key, region)
+      container_engine_client = OCI::ContainerEngine::ContainerEngineClient.new(:config => config)
+
+      # Prepare a signed request
+      signer = container_engine_client.api_client.instance_variable_get(:@signer)
+
+      url = URI::HTTPS.build(
+        :host => "containerengine.#{region}.oraclecloud.com",
+        :path => "/cluster_request/#{cluster_id}"
+      )
+
+      params = {}
+      signer.sign(:GET, url, params, nil)
+
+      url.query = params.to_query
+
+      Base64.urlsafe_encode64(url.to_s)
+    end
+
+    private
+
+    def oci_config(tenant, user, private_key, public_key, region)
```

Comment:
TODO this is the same code as `ManagerMixin.raw_connect` and should be refactored